shared/src/ledger/queries_ext.rs

```diff
@@ -433,4 +439,14 @@ where
             },
         ))
     }
+
+    fn get_max_proposal_bytes(&self) -> ProposalBytes {
+        let key = get_max_proposal_bytes_key();
+        let (maybe_value, _gas) = self
+            .read(&key)
+            .expect("Must be able to read ProposalBytes from storage");
+        let value =
+            maybe_value.expect("ProposalBytes must be present in storage");
+        decode(value).expect("Must be able to decode ProposalBytes in storage")
+    }
```

Comment:
I still think ideally we should be returning errors rather than panicking here. The issue is we do not have a nice way to gracefully shut down within Shell ABCI methods upon encountering an unrecoverable error, so we will have to panic somewhere up the stack right now anyway.

Reply:
A panic in the shell triggers the abortable spawner which shuts down all the other processes. I think we should panic here.